tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Consider adding a test case for handling multiple syntax errors in a single file.

This test currently covers the scenario where a file with a syntax error is processed. It would be beneficial to also test how the extractor behaves when multiple syntax errors are present in a single file, ensuring that all errors are appropriately logged or warned.